### ISSUE TITLE
Compatibility with 0.3 for qr

### ISSUE THREAD

**Issue by martinholters** [JuliaLang/Compat.jl#160]
I have recently added

```Julia
if VERSION < v"0.4.0-dev+2840"
    qr(A, ::Type{Val{true}}; thin::Bool=true) =
        Base.qr(A, pivot=true, thin=thin)
    qr(A, ::Type{Val{false}}; thin::Bool=true) =
        Base.qr(A, pivot=false, thin=thin)
end
```

to a project. This looks like a reasonable addition for Compat, but I'm not sure whether this is better handled using expression rewriting with `@compat` like what is done for `chol`.

Anyway, having compatibility for `qr` in Compat would be nice.

**Comment by andreasnoack** [comment]
I think this is fine. It would be great if you could open a PR.

**Comment by yuyichao** [comment]
Closed by #161
